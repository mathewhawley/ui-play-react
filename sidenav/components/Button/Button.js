import React, { PropTypes } from 'react';
import classnames from 'classnames';
import styles from './Button.scss';

const Button = ({
  onClick,
  icon,
  customStyles,
}) => {
  const classes = classnames(styles.base, customStyles, 'material-icons');
  return (
    <button className={classes} onClick={onClick}>{icon}</button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  icon: PropTypes.string.isRequired,
  customStyles: PropTypes.string,
};

export default Button;
