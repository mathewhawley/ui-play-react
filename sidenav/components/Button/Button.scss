.base {
  min-width: 48px;
  min-height: 48px;
}
