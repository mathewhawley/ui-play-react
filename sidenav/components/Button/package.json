{
  "private": true,
  "name": "Button",
  "main": "./Button.js"
}
