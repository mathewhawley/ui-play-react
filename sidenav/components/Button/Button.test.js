/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('<Button />', () => {
  let wrapper;
  let clickHandler;

  beforeEach(() => {
    clickHandler = jest.fn();
    wrapper = shallow(<Button onClick={clickHandler} icon='menu' />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should call handler on click event', () => {
    wrapper.simulate('click');
    expect(clickHandler).toHaveBeenCalled();
  });
});
