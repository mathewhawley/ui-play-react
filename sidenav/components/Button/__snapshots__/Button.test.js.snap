exports[`<Button /> renders correctly 1`] = `
<button
  className="base material-icons"
  onClick={[Function]}>
  menu
</button>
`;
